demos/object_detection_demo_ssd_async/main.cpp

```diff
@@ -50,523 +135,127 @@ bool ParseAndCheckCommandLine(int argc, char *argv[]) {
-        InputsDataMap inputInfo(cnnNetwork.getInputsInfo());
-
-        std::string imageInputName, imageInfoInputName;
-        size_t netInputHeight, netInputWidth;
-
-        for (const auto& inputInfoItem : inputInfo) {
-            if (inputInfoItem.second->getTensorDesc().getDims().size() == 4) {  // 1st input contains images
-                imageInputName = inputInfoItem.first;
-                inputInfoItem.second->setPrecision(Precision::U8);
-                if (FLAGS_auto_resize) {
-                    inputInfoItem.second->getPreProcess().setResizeAlgorithm(ResizeAlgorithm::RESIZE_BILINEAR);
-                    inputInfoItem.second->getInputData()->setLayout(Layout::NHWC);
-                } else {
-                    inputInfoItem.second->getInputData()->setLayout(Layout::NCHW);
-                }
-                const TensorDesc& inputDesc = inputInfoItem.second->getTensorDesc();
-                netInputHeight = getTensorHeight(inputDesc);
-                netInputWidth = getTensorWidth(inputDesc);
-            } else if (inputInfoItem.second->getTensorDesc().getDims().size() == 2) {  // 2nd input contains image info
-                imageInfoInputName = inputInfoItem.first;
-                inputInfoItem.second->setPrecision(Precision::FP32);
-            } else {
-                throw std::logic_error("Unsupported " +
-                                       std::to_string(inputInfoItem.second->getTensorDesc().getDims().size()) + "D "
-                                       "input layer '" + inputInfoItem.first + "'. "
-                                       "Only 2D and 4D input layers are supported");
-            }
-        }
-
-        // --------------------------- Prepare output blobs -----------------------------------------------------
-        slog::info << "Checking that the outputs are as the demo expects" << slog::endl;
-        OutputsDataMap outputInfo(cnnNetwork.getOutputsInfo());
-        if (outputInfo.size() != 1) {
-            throw std::logic_error("This demo accepts networks having only one output");
-        }
-        DataPtr& output = outputInfo.begin()->second;
-        auto outputName = outputInfo.begin()->first;
-
-        int num_classes = 0;
-
-        if (auto ngraphFunction = cnnNetwork.getFunction()) {
-            for (const auto op : ngraphFunction->get_ops()) {
-                if (op->get_friendly_name() == outputName) {
-                    auto detOutput = std::dynamic_pointer_cast<ngraph::op::DetectionOutput>(op);
-                    if (!detOutput) {
-                        THROW_IE_EXCEPTION << "Object Detection network output layer(" + op->get_friendly_name() +
-                            ") should be DetectionOutput, but was " +  op->get_type_info().name;
-                    }
-
-                    num_classes = detOutput->get_attrs().num_classes;
+                frameNum = pipeline.submitImage(curr_frame);
+                if (frameNum < 0)
```

Comment:
You've already checked that `pipeline` can take the frame by `isReadyToProcess()`. Remove the check.

Reply:
Fixed